### PR TITLE
gh-102823: Document return type of floor division on floats

### PR DESCRIPTION
**Pull request by mdickinson**
This PR documents the long-standing behaviour that the floor division operator `//` applied to `float` operands returns a result of type `float`.

Also adds a missing note to the table to note that `x // y` isn't valid for complex numbers.

* Issue: gh-102823

### CODE REVIEW

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -319,7 +319,9 @@ the operations, see :ref:`operator-summary`):
 Notes:
 
 (1)
-   Also referred to as integer division.  The resultant value is a whole
+   Also referred to as integer division.  For operands of type :class:`int`,
+   the result has type :class:`int`.  For operands of type :class:`float`,
```
Comment (Eclips4) [inline Doc/library/stdtypes.rst:323]: I think, some number of readers can understand this as "If both operands are float, the result has type float".
But, in fact, if one of operands are float, the result type is float.

File: Doc/library/stdtypes.rst [hunk 2]
```diff
@@ -319,7 +319,9 @@ the operations, see :ref:`operator-summary`):
...
+   the result has type :class:`float`.  In general, the result is a whole
```
Comment (skirpichev) [inline Doc/library/stdtypes.rst:324]:
```suggestion
   Also referred to as integer division.  If both operands of type :class:`int`,
   the result has type :class:`int`.  If any operand of type :class:`float`,
   the result has type :class:`float`.  In general, the result is a whole
```

File: Doc/library/stdtypes.rst [hunk 3]
```diff
@@ -282,7 +282,7 @@ the operations, see :ref:`operator-summary`):
 +---------------------+---------------------------------+---------+--------------------+
 | ``x / y``           | quotient of *x* and *y*         |         |                    |
 +---------------------+---------------------------------+---------+--------------------+
-| ``x // y``          | floored quotient of *x* and     | \(1)    |                    |
+| ``x // y``          | floored quotient of *x* and     | \(1)\(2)|                    |
```
Comment (skirpichev) [inline Doc/library/stdtypes.rst:285]: Just an unrelated nitpick: I wonder why this table was formatted this way?  Couldn't we use sphinx's footnotes?

File: Doc/library/stdtypes.rst (hunk 2 above)
Comment (skirpichev) [reply Doc/library/stdtypes.rst:324]: Actually, this turns to be more complicated, if we consider all numeric types of the stdlib.

It seems, the result is an int if _both_ operands are of exact type (the Scheme numerical tower has this notion), i.e. either int or Fraction.  But if any operand is a float or a Decimal - that defines the type of the result (if one defined for such combination).

File: Doc/library/stdtypes.rst (hunk 1 above)
Comment (mdickinson) [reply Doc/library/stdtypes.rst:323]: > But, in fact, if one of operands are float, the result type is float.

This is already covered earlier in the docs where it talks about mixed arithmetic. There isn't room for an exhaustive combinatorial description here.

Similarly, let's leave details about `Fraction` and `Decimal` to the respective module documentation. All we need to cover here is `int`, `float` and `complex`.

File: Doc/library/stdtypes.rst (hunk 2 above)
Comment (mdickinson) [reply Doc/library/stdtypes.rst:324]: Let's leave details about `Fraction` and `Decimal` to the respective module documentation; I don't think this is the place to go through all the possible combinations.

File: Doc/library/stdtypes.rst (hunk 3 above)
Comment (mdickinson) [reply Doc/library/stdtypes.rst:285]: I suggest opening a separate issue if you want to pursue this; it's out of scope for this PR.

File: Doc/library/stdtypes.rst (hunk 1 above)
Comment (skirpichev) [reply Doc/library/stdtypes.rst:323]: > This is already covered earlier in the docs where it talks about mixed arithmetic.

Indeed, right above the table ("when a binary arithmetic operator has operands of different numeric types, the operand with the “narrower” type is widened to that of the other...").

**Review by skirpichev — APPROVED** [review]

**Review by Eclips4 — APPROVED** [review]
Indeed, information about mixed arithmetic is given above.
LGTM

File: Doc/library/stdtypes.rst [hunk 4]
```diff
@@ -319,7 +319,9 @@ the operations, see :ref:`operator-summary`):
 Notes:
 
 (1)
-   Also referred to as integer division.  The resultant value is a whole
+   Also referred to as integer division.  For operands of type :class:`int`,
+   the result has type :class:`int`.  For operands of type :class:`float`,
+   the result has type :class:`float`.  In general, the result is a whole
    integer, though the result's type is not necessarily int.  The result is
```
Comment (hugovk) [inline Doc/library/stdtypes.rst]:
```suggestion
   integer, though the result's type is not necessarily :class:`int`.  The result is
```

**Review by hugovk — APPROVED** [review]
Thanks!

**Pull request merged**